src/services/EcoBridge/EcoBridge.providers.ts

```diff
@@ -0,0 +1,31 @@
+import { ChainId } from '@swapr/sdk'
+import { INFURA_PROJECT_ID } from '../../connectors'
+import { NETWORK_DETAIL } from '../../constants'
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { EcoBridgeProviders } from './EcoBridge.types'
+
+const addInfuraKey = (rpcUrl: string) => {
+  if (rpcUrl.includes('infura')) {
+    let updatedUrl = rpcUrl
+
+    if (!rpcUrl.endsWith('/')) {
+      updatedUrl = rpcUrl + '/'
+    }
```

Comment:
Do you want to make this into a ternary operator?
Or any logic that makes line 12 build on top of line 9 instead of overwriting it.

Reply:
Resolved by #900